aea/helpers/base.py

```diff
@@ -68,3 +70,13 @@ def locate(path):
         except AttributeError:
             return None
     return object
+
+
+def generate_public_id(user_name: str, package_name: str, version: str, nonce: Optional[int] = None) -> str:
+    """Generate a unique id for the package."""
+    import hashlib
+    string_for_hash = "".join([user_name, package_name, version, str(nonce)])
+    m_hash = hashlib.sha3_256()
+    m_hash.update(string_for_hash.encode())
+    encoded_str = m_hash.digest().hex()
+    return encoded_str
```

Comment:
So what do we have on the output? Hashed string?

Reply:
yeap